src/toil/batchSystems/slurm.py

```diff
@@ -444,6 +443,29 @@ def prepareSbatch(self,
             if cpu is not None:
                 sbatch_line.append(f'--cpus-per-task={math.ceil(cpu)}')
 
+            if gpus:
+                # This block will add a gpu supported partition only if no partition is supplied by the user
+                sbatch_line = sbatch_line[:1] + [f'--gres=gpu:{gpus}'] + sbatch_line[1:]
+                lowest_priority_gpus = sorted([i for i in self.boss.partitions if i["gres"]], key=lambda x: x["priority"])
+                new_partition_name = lowest_priority_gpus[0][0]  # get name of lowest priority partition that supports gpus
+                if not any(option.startswith("--partition") for option in sbatch_line):
+                    # no partition specified, so specify one
+                    sbatch_line.append(f"--partition={new_partition_name}")
+                else:
+                    # there is a partition specified already, check if the partition has GPUs
+                    for i, option in enumerate(sbatch_line):
+                        if option.startswith("--partition"):
+                            # grab the partition name depending on if it's specified via an "=" or a space
+                            if "=" in option:
+                                partition_name = option[len("--partition="):]
+                            else:
+                                partition_name = option[i+1]
+
+                            if not any(partition_name == partition for partition in lowest_priority_gpus):
+                                # the specified partition is not compatible, so override with a gpu supported partition
+                                logger.warning(f"Job {jobName} needs {gpus} GPUs, but specified partition {partition_name} is incompatible. This job may not work.")
```

Comment:
If we do know of a partition with GPUs, maybe we should name drop it here to help the user.